aws/nextflow-compute.tf

```diff
@@ -0,0 +1,56 @@
+resource "aws_iam_instance_profile" "nf_ecs_instance_role" {
+  name = "nextflow-ecs-instance-role"
+  role = aws_iam_role.nf_ecs_role.name
+}
+
+resource "aws_batch_compute_environment" "nextflow_ondemand" {
+  compute_environment_name = "nextflow-ondemand-compute"
+  compute_resources {
+    instance_role = aws_iam_instance_profile.nf_ecs_instance_role.arn
+    instance_type = [
+      "optimal",
+    ]
+    allocation_strategy = "BEST_FIT"
+    max_vcpus = 20
+    min_vcpus = 0
+    image_id = "ami-0a8857ac38c35157f"
+    security_group_ids = [
+      aws_security_group.nf_security.id,
+    ]
+    subnets = [
+      aws_subnet.nf_subnet.id,
+    ]
+    type = "EC2"
+  }
+
+  service_role = aws_iam_role.nf_batch_role.arn
+  type         = "MANAGED"
+  depends_on   = [aws_iam_role_policy_attachment.nf_batch]
+}
+
+resource "aws_batch_compute_environment" "nextflow_spot" {
+  compute_environment_name = "nextflow-spot-compute"
+  compute_resources {
+    instance_role = aws_iam_instance_profile.nf_ecs_instance_role.arn
+    instance_type = [
+      "optimal",
+    ]
+    allocation_strategy = "SPOT_CAPACITY_OPTIMIZED"
+    spot_iam_fleet_role = aws_iam_role.nf_ecs_role.arn
+    bid_percentage = 20
+    max_vcpus = 100
+    min_vcpus = 0
```

Comment:
Ok so if I understand this correctly, you've got a default queue that uses spot instances and a priority queue that uses on demand instances. They're both going to scale up automatically, default to a max_vcpus of 100 and the priority to 20. Is that all correct?

Reply:
That is what is supposed to happen, yes. I may well change these in the future, but this seemed a good place to start.

Reply:
Ok sweet. A summary like that might also be useful to have in a comment or doc somewhere.

Reply:
Yeah I was planning to add a readme before making it into a real PR.